pints/toy/stochastic/_markov_jump_model.py

```diff
@@ -0,0 +1,149 @@
+#
+# Markov jump model.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+from scipy.interpolate import interp1d
+import pints
+import random
+
+from .. import ToyModel
+
+
+class MarkovJumpModel(pints.ForwardModel, ToyModel):
+    r"""
+    A general purpose Markov Jump model used for any systems of reactions
+    that proceed through jumps. We simulate a population of N different species
+    reacting through M different mechanisms
+
+    Simulations are performed using Gillespie's algorithm [1]_, [2]_:
+
+    1. Sample values :math:`r_0`, :math:`r_1`, from a uniform distribution
+
+    .. math::
+        r_0, r_1 \sim U(0,1)
+
+    2. Calculate the time :math:`\tau` until the next single reaction as
+
+    .. math::
+        \tau = \frac{-\ln(r)}{a_0}
+
+    3. Decide which reaction, i, takes place using r_1 * a_0 and iterating
+
+    through propensities
```

Comment:
This needs quite a bit more explanation, especially since this is the key point of the algorithm